src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -36,9 +36,9 @@ void CoreWorkerDirectTaskSubmitter::AddWorkerLeaseClient(
                                   std::make_pair(std::move(lease_client), expiration));
 }
 
-void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(const rpc::WorkerAddress &addr,
-                                                 const SchedulingKey &scheduling_key,
-                                                 bool was_error) {
+void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(
+    const rpc::WorkerAddress &addr, const SchedulingKey &scheduling_key, bool was_error,
+    const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> resource_mapping) {
```

Comment:
```suggestion
    const google::protobuf::RepeatedPtrField<rpc::ResourceMapEntry> &assigned_resources) {
```
I'm not sure if I'm understanding this code correctly, but assigned_resources seems accurate.